product/static/templates/products/user_products.js

```diff
@@ -0,0 +1,41 @@
+function product_html(data) {
+    return ('<tr id=' + data.id + '>' +
+    '<td><input class = "name" type="text" readonly value=' + data.Name + '></td> ' +
+    '<td><input class="detail" type="text" readonly value=' + data.Detail + '></td>' +
+    '<td><input class="price" type="text" readonly value=' + data.Price + '></td>' +
+    '</tr>');
+}
+function append_product(data, append) {
+    var html_string = product_html(data);
+    append == undefined ? $("#main").append(html_string) : $("#main").html(html_string);
```

Comment:
Add this  logic to a function and call this function everywhere you need to to add or append the html.
Also there is no need of equality check for `undefined`. you can use just `append?true path:false path` 
or `if (append)`